osmosis-labs/osmosis-frontend
# (CL) Swap tool promo

## What is the purpose of the change

Adds a conditional promo drawer that animates above and behind the swap tool of one of the selected currencies is in _a_ CL pool. The button provides a link to the pools page with the concentrated pool filter in the URL params.

### ClickUp Task

Please provide a link to the ClickUp task for this pull request:

[ClickUp Task URL](https://app.clickup.com/t/8677q2nvp)

## Brief Changelog

<img width="464" alt="Screenshot 2023-06-03 at 6 34 23 AM" src="https://github.com/osmosis-labs/osmosis-frontend/assets/4606373/461faa0c-a78d-40a7-967b-08610391cd6f">

- Adds animated background drawer to swap too iff one of the assets in the swap tool is contained with >=1 CL pool in the swap tool

## Key Workflows to test

- Select a CL asset, observe the drawer animate up
- Navigate to and from other pages, notice the drawer appear if the assets are still in a CL pool

## Testing and Verifying

Make sure the drawer only appears if a CL pool contains the selected asset.

### Figma
https://www.figma.com/file/RxNOvfvzmgtsgCPUetQvcG/Supercharged-Liquidity?type=design&node-id=1173-40832&t=gbUrjyXIcQuVLhgB-4
